### PR TITLE
chore: Upgrade react, react-dom, react-native to latests versions

### PR DESCRIPTION
**Pull request by machour**
## Description

Resurrecting #523, I updated react, react-native to their latests versions using `react-native-git-upgrade`. I then proceeded to fix conflicts and get rid of `index.ios.js` & `index.android.js` in favor of `index.js` 🎉 

Tested the application on both iOS and Android simulator, nothing seems to brake.

The app startup is quicker, it becomes responsive as soon as it's launched, overall, it seems to be on steroïds ... WOW.

I was only able to validate this on Android, as I never was able to run the app on iOS without hardcoding my IP in `AppDelegate.m`. @chinesedfan could you give it a try on iOS ?

For this PR, the best thing is to `rm -rf node_modules` and then `yarn && yarn run link && yarn start --reset-cache`

PS: Upgrading react-native-linear-gradient is mandatory as it fixes a compatibility problem: react-native-community/react-native-linear-gradient#235

### CODE REVIEW

**Comment by machour** [comment]
Please do \o/

File: ios/GitPoint/AppDelegate.m [hunk 1]
```diff
@@ -20,11 +20,11 @@ - (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(
...
   NSURL *jsCodeLocation;
 
-
+  
 #ifdef DEBUG
-  jsCodeLocation = [[RCTBundleURLProvider sharedSettings] jsBundleURLForBundleRoot:@"index.ios" fallbackResource:nil];
+    jsCodeLocation = [[RCTBundleURLProvider sharedSettings] jsBundleURLForBundleRoot:@"index" fallbackResource:nil];
 #else
-  jsCodeLocation = [CodePush bundleURL];
+    jsCodeLocation = [CodePush bundleURL];
```
Comment (chinesedfan) [inline ios/GitPoint/AppDelegate.m]: Shall we keep the pure empty line and old indentations?

**Comment by machour** [comment]
@chinesedfan Fixed the whitespaces changes
@housseindjirdeh I'd love it if you can give this a go and merge this one in <3

File: ios/GitPoint/AppDelegate.m [hunk 2]
```diff
@@ -20,9 +20,9 @@ - (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(
 {
   NSURL *jsCodeLocation;
 
-
+
```
Comment (chinesedfan) [inline ios/GitPoint/AppDelegate.m]: > the pure empty line

@machour Sorry, I didn't comment very clearly before.

Comment (machour) [reply ios/GitPoint/AppDelegate.m]: Originally there was a line with a single space in it. I just removed it all together.
Not sure if I'm following you though :D

**Pull request merged**